src/event/writer.rs

```diff
@@ -137,10 +154,22 @@ impl EventWriter {
     ) -> oneshot::Receiver<Result<(), Error>> {
         let size = event.len();
         let (tx, rx) = oneshot::channel();
+        let (tx_flush, rx_flush) = oneshot::channel();
         let routing_info = RoutingInfo::RoutingKey(Some(routing_key));
+        let routing_info_flush = RoutingInfo::RoutingKey(None);
```

Comment:
This will route the flush event differently from the event it is supposed to be following. This could result in flush returning even when the data has not yet been written. Please add a test that covers this case.